src/main/groovy/com/twcable/grabbit/jcr/JcrNodeDecorator.groovy

```diff
@@ -135,6 +139,66 @@ class JcrNodeDecorator {
         }
     }
 
+    private void checkoutNode() {
+        try {
+            JcrNodeDecorator decoratedVersionableAncestor = findVersionableAncestor();
+            if (decoratedVersionableAncestor && !decoratedVersionableAncestor.isCheckedOut()) {
+                decoratedVersionableAncestor.session.workspace.versionManager.checkout(decoratedVersionableAncestor.path);
+            }
+        }
+        catch (Exception exception) {
+            log.warn "Could not checkout node ${this.path}"
+        }
+    }
+
+    private JcrNodeDecorator findVersionableAncestor() throws RepositoryException {
+        if (isVersionable()) {
+            return this;
+        }
+        try {
+            JcrNodeDecorator parentDecoratedNode = new JcrNodeDecorator(this.parent);
+            return parentDecoratedNode.findVersionableAncestor();
+        } catch (ItemNotFoundException e) {
+            return null;
+        }
+    }
+
+    @CompileStatic(TypeCheckingMode.SKIP)
+    private boolean isVersionable() throws RepositoryException {
+        return mixinNodeTypes.any { it.name in ["mix:simpleVersionable", MIX_VERSIONABLE] }
+    }
+
+    /*
+    * Write proto properties to Jcr Node
+    * Mixins are added before other properties
+    */
+
+    void writeToJcr(ProtoNodeDecorator decoratedProtoNode, @Nonnull Session session) {
```

Comment:
I'm confused. This is a JcrNodeDecorator. It wraps a node that is already within the JCR. Why would we call writeToJcr on an instance of this object and pass a ProtoNodeDecorator?